packages/overlay/src/OverlayTrigger.ts

```diff
@@ -179,26 +179,28 @@ export class OverlayTrigger extends LitElement {
         }
     }
 
-    private onClickSlotChange(event: Event): void {
+    private onClickSlotChange(
+        event: Event & { target: HTMLSlotElement }
```

Comment:
I've never seen this typing before. Looks very useful

Reply:
👍 I'm still trying to understand why _sometimes_ TS will be annoyed by this, but in places where we can this reduces boilerplate internal of the event listener that is harder to be washed out by the TS build.